common/src/binary/BinaryBuffer.cpp

```diff
@@ -0,0 +1,105 @@
+#include <rcbe-engine/binary/BinaryBuffer.hpp>
+
+namespace rcbe::binary {
+
+template <>
+BinaryBuffer::BinaryBuffer(size_t v) {
+    buffer_.resize(sizeof(size_t));
+    auto * _v = reinterpret_cast<char*>(&v);
+    std::copy(_v, _v + sizeof(v), buffer_.data());
+}
+
+template <>
+BinaryBuffer::BinaryBuffer(float f) {
+    buffer_.resize(sizeof(float));
+    auto * _v = reinterpret_cast<char*>(&f);
+    std::copy(_v, _v + sizeof(float), buffer_.data());
+}
+
+template <>
+BinaryBuffer::BinaryBuffer(char c) {
+    buffer_.resize(sizeof(char));
+    auto * _v = reinterpret_cast<char*>(&c);
+    std::copy(_v, _v + sizeof(char), buffer_.data());
+}
+
+template <>
+BinaryBuffer::BinaryBuffer(uint32_t c) {
+    buffer_.resize(sizeof(uint32_t));
+    auto * _v = reinterpret_cast<uint32_t*>(&c);
+    std::copy(_v, _v + sizeof(uint32_t), buffer_.data());
+}
+
+template <>
+BinaryBuffer::BinaryBuffer(uint16_t c) {
+    buffer_.resize(sizeof(uint16_t));
+    auto * _v = reinterpret_cast<uint16_t*>(&c);
+    std::copy(_v, _v + sizeof(uint16_t), buffer_.data());
+}
+
```

Comment:
Looks like above implementations can be made into a single function.

Reply:
Well not quite, this specializations of template functions if done the way you suggested would need to call another template function, which would lead to code bloat actually. So i think these should be left as is.

Reply:
'Kay